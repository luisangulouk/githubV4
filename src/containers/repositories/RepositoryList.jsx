/* eslint-disable no-unused-vars */
/* eslint-disable no-underscore-dangle */
/* eslint-disable no-console */
import React from 'react';
import Style from './repositories.scss';

const RepositoryList = ({
  repositories
}) => {
  console.log('repositories:', repositories);
  return (
    <div className={Style.repoWrapper}>
      <h5 className={Style.repoHeader}>Latest Repositories created by this user</h5>
      <ul className={Style.repoList}>
        {repositories.edges.map(({node}) => {
          console.log('node:', node);
          return (
            <li key={node.id} className={Style.repoProfile}>
              <span className={Style.repoName}>{node.name}</span><br />
              <span className={Style.repoCreatedAt}>{node.createdAt}</span><br />
              {node.languages.nodes && node.languages.nodes.map(lang => {
                return <span key={lang.name} className={`badge badge-info ${Style.repoLanguage}`}>{lang.name}</span>;
              })}
            </li>
          );
        }
        )}
      </ul>
    </div>
  );
};

export default RepositoryList;
