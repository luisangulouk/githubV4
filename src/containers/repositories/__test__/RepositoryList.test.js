import React from 'react';
import renderer from 'react-test-renderer';

import RepositoryList from '../RepositoryList';

describe('Repository List Component', () => {

  const mockRepositories = {
    search: {
      nodes: [{
        createdAt: '2016-09-06T15:46:44Z',
        id: 'MDEwOlJlcG9zaXRvcnk2NzUyMjU3Nw==',
        languages: {
          nodes: [{
            name: 'C#',
            __typename: 'Language'
          }],
          __typename: 'LanguageConnection'
        },
        name: 'XamarinSQL',
        nameWithOwner: 'icebeam7/XamarinSQL',
        refs: {
          nodes: [{
            id: 'MDM6UmVmNjc1MjI1Nzc6cmVmcy9oZWFkcy9tYXN0ZXI=',
            name: 'master',
            __typename: 'Ref'
          }],
          __typename: 'RefConnection'
        },
        __typename: 'Repository'
      }],
      __typename: 'SearchResultItemConnection'
    }
  };

  it('should match snapshot', () => {

    const component = renderer.create(<RepositoryList repositories={mockRepositories} login={'icebeam7'} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
