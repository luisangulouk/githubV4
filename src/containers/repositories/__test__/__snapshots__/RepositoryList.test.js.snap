// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repository List Component should match snapshot 1`] = `
<div
  className="repoWrapper"
>
  <h5
    className="repoHeader"
  >
    Repositories found in this colletion are not Public
  </h5>
  <ul
    className="repoList"
  />
</div>
`;
